### PR TITLE
Issue #560 Remove setup.cfg

### PR DESCRIPTION
**Pull request by Manangka**
In this commit all the contents of setup.cfg has been moved to the pyproject.toml file.
Flake8 doesn't support the pyproject.toml format, therefor it has been relaced with ruff.

### CODE REVIEW

**Comment by Huite** [comment]
I would recommend moving to ruff instead and totally getting rid of flake.

It's likely that ruff is a bit stricter on a number of fields. I've been using the config that I think @Hofer-Julian set up for Ribasim with good effect: 

See e.g.:
https://github.com/Deltares/xugrid/blob/main/ruff.toml

**Comment by Manangka** [comment]
Alright, I will have a look at Ruff

**Comment by Hofer-Julian** [comment]
Ruff simply reimplements flake8, isort, black and many others. But it's much faster and you only need to install one thing

File: pyproject.toml [hunk 1]
```diff
@@ -2,13 +2,137 @@
...
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering :: Hydrology",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (JoerivanEngelen) [inline pyproject.toml:29]: This is outdated info, should be:

```suggestion
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
```

File: pyproject.toml [hunk 2]
```diff
@@ -2,13 +2,137 @@
...
+    "tqdm",
+    "xarray >=0.15",
+    "xugrid >=0.3.0",
+]
+
+[project.urls]
+Homepage = "https://github.com/Deltares/imod-python"
+Documentation = "https://deltares.github.io/imod-python/"
+Repository = "https://github.com/Deltares/imod-python.git"
+"Bug Tracker" = "https://github.com/Deltares/imod-python/issues"
```
Comment (JoerivanEngelen) [inline pyproject.toml:56]: Why the quotes around "Bug Tracker"?

**Review by JoerivanEngelen — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (Manangka) [reply pyproject.toml:55]: I followed the example on:
https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#advanced-plugins

In that example the "Bug Tracker" is also in quotes

**Review by JoerivanEngelen — APPROVED** [review]

**Pull request merged**